log4j-core/src/main/java/org/apache/logging/log4j/core/layout/AbstractStringLayout.java

```diff
@@ -270,9 +265,10 @@ protected String serializeToString(final Serializer serializer) {
         if (serializer == null) {
             return null;
         }
-        final LoggerConfig rootLogger = getConfiguration().getRootLogger();
+        final LoggerConfig rootLogger = configuration.getRootLogger();
+        final LogEventFactory logEventFactory = configuration.getLogEventFactory();
         // Using "" for the FQCN, does it matter?
-        final LogEvent logEvent = getLogEventFactory().createEvent(rootLogger.getName(), null, Strings.EMPTY,
+        final LogEvent logEvent = logEventFactory.createEvent(rootLogger.getName(), null, Strings.EMPTY,
```

Comment:
I don't see the need for extracting a local variable.

Reply:
The getter wasn't for a field; it was doing work each time. I realized upon fixing this that `Configuration` must be non-null at this point, so we can just get it from `Configuration::getLogEventFactory` (recently added method).

Reply:
Agreeing with @jvz here. If a method is not merely a simple field getter (which is a shame that it is still called `getBlah`), its result should first better be stored in a self-explanatory variable. This greatly helps while stepping in a debugger.